test/sanity/code-smell/docs-build.requirements.txt

```diff
@@ -1,11 +1,11 @@
 jinja2 == 3.0.1
 pyyaml == 5.4.1
 resolvelib == 0.5.4
-sphinx == 2.1.2
-sphinx-notfound-page == 0.7.1
+sphinx == 4.2.0
+sphinx-notfound-page == 0.8
```

Comment:
The sphinx requirements for `rstcheck` should be updated to match as well:

https://github.com/ansible/ansible/blob/devel/test/sanity/code-smell/rstcheck.requirements.txt